ios/gamma/stack/RNSStackScreenComponentView.mm

```diff
@@ -116,6 +138,34 @@ + (BOOL)shouldBeRecycled
   return NO;
 }
 
+#ifdef RCT_NEW_ARCH_ENABLED
+
+#pragma mark - RCTMountingTransactionObserving
+
+- (void)mountingTransactionWillMount:(const facebook::react::MountingTransaction &)transaction
+                withSurfaceTelemetry:(const facebook::react::SurfaceTelemetry &)surfaceTelemetry
+{
+  for (const auto &mutation : transaction.getMutations()) {
+    if ([self shouldInvalidateOnMutation:mutation]) {
+      [RNSViewControllerInvalidator invalidateViewIfDetached:self forRegistry:_invalidatedComponentsRegistry];
+    }
+  }
+}
```

Comment:
First thing is: why do we have to scan whole list of mutations *in every screen component*? Why can't we just do it once per transaction in host component?

Reply:
🤔 I don't have good suggestion here unfortunately, but I dislike this solution for it's excessive (hopefully) transaction scanning. I'll try to come up with something better soon. We can discuss this on Monday.